src/viewer.js

```diff
@@ -124,14 +139,49 @@ class Viewer {
   }
   
   /**
-   * @param {?string} urlPath
+   * @param {boolean} isLastBack true if back button was hit and viewer should hide.
+   * @param {boolean} isAMP true if going to AMP document.
+   * @param {number|undefined} opt_poppedAmpHistoryIndex The AMP History Index that was just popped.
    * @private
     */
-  handleChangeHistoryState_(urlPath) {
-    if (urlPath) {
-      if (this.showViewer_) this.showViewer_();
-    } else {
+  handleChangeHistoryState_(isLastBack, isAMP, opt_poppedAmpHistoryIndex) {
+    if (isLastBack) {
       if (this.hideViewer_) this.hideViewer_();
+      return;
+    }
+    if (isAMP && this.showViewer_ && this.isViewerHidden_ && this.isViewerHidden_()) {
+      this.showViewer_();
+      return;
+    }
+    this.viewerMessaging_.sendRequest(
+      'historyPopped', 
+      // The back button was hit while a lightbox or sidebar is open.
+      // We subtract 1 from opt_poppedAmpHistoryIndex to tell the AMP
+      // page to go back to the previous state (close the lightbox/sidebar).
+      {'newStackIndex': opt_poppedAmpHistoryIndex - 1}, 
```

Comment:
I don't think you want to subtract 1. The state should be the newly active state, which is what you want to provide to the document.